src/NServiceBus.Core/Receiving/ReceiveComponent.cs

```diff
@@ -0,0 +1,93 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using ObjectBuilder;
+    using Routing;
+    using Settings;
+    using Transport;
+
+    class ReceiveComponent
+    {
+        public ReceiveComponent(string endpointName, bool isSendOnlyEndpoint, TransportInfrastructure transportInfrastructure)
+        {
+            this.endpointName = endpointName;
+            this.isSendOnlyEndpoint = isSendOnlyEndpoint;
+            this.transportInfrastructure = transportInfrastructure;
+        }
+
+        public ReceiveConfiguration Configure(ReadOnlySettings settings)
+        {
+            if (isSendOnlyEndpoint)
+            {
+                return new ReceiveConfiguration(new LogicalAddress(), null, null, null, TransportTransactionMode.None, PushRuntimeSettings.Default, false, false);
```

Comment:
I don't like the "send-only" receive configuration concept. I think it should be null so that it is more explicit and pushes un in the direction of having a separate top-level "thing" for send-only endpoints.

But I guess the next step could be to create an explicit `SendingComponent` which, ideally, should also be runnable without the whole endpoint.

Reply:
Yea, the send only stuff is a huge mess. This was the best I could to not expand the scope to much. I'll try the null idea and see what it breaks